src/App.php

```diff
@@ -161,6 +161,11 @@ function () use ( $slim ) {
 					$page->setDao( $slim->plagiabotDao );
 					$page->setEnwikiDao( $slim->enwikiDao );
 					$page->setWikiprojectDao( $slim->wikiprojectDao );
+
+					// determine if we are on the staging environment, so we can show a banner in the view
+					$rootUri = $slim->request->getRootUri();
+					$slim->view->set( 'staging', preg_match( '/\/plagiabot/', $rootUri ) );
```

Comment:
Not exactly, but I'll take care of this now and merge in a bit. Because we might need to do this in several places, it makes more sense to add it to the middleware array above (L143) and then simply include it, like I've done with inject-user.